### ISSUE TITLE
For cyclic learning rate, update LR every iteration, instead of every epoch

### ISSUE THREAD

**Issue by epwalsh** [epwalsh/pytorch-crf#85]
- [x] update train.py
- [x] remove vanilla decay, only use cyclic decay
- [x] update cyclic decay
- [x] update command line options
- [x] update test
- [x] add test when training size is set
- [x] add test for when cycle_len is equal to epochs
- [x] try on small sample dataset
